examples/ai-core/src/e2e/google-vertex.test.ts

```diff
@@ -453,5 +455,38 @@ describe.each(Object.values(RUNTIME_VARIANTS))(
         expect(result.usage?.tokens).toBeGreaterThan(0);
       });
     });
+
+    describe.each(MODEL_VARIANTS.image)('Image Model: %s', modelId => {
+      it('should generate an image with correct dimensions and format', async () => {
+        const model = vertex.image(modelId);
+        const { image } = await generateImage({
+          model,
+          prompt: 'A burrito launched through a tunnel',
+          size: '1024x1024',
+        });
+
+        // Verify we got a Uint8Array back
+        expect(image.uint8Array).toBeInstanceOf(Uint8Array);
+
+        // Check the file size is reasonable (at least 10KB, less than 10MB)
+        expect(image.uint8Array.length).toBeGreaterThan(10 * 1024);
+        expect(image.uint8Array.length).toBeLessThan(10 * 1024 * 1024);
+
+        // Verify PNG format by checking magic numbers
+        const pngSignature = [137, 80, 78, 71, 13, 10, 26, 10];
```

Comment:
we have a mime type detection helper for this but I'm not sure it's exposed

Reply:
Exposed it for use, much nicer.